chain/consensus/platform/logging/logger.go

```diff
@@ -0,0 +1,51 @@
+package logging
+
+import (
+	"context"
+
+	logging "github.com/ipfs/go-log/v2"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type contextKey string
+
+const loggerKey = contextKey("logger")
+
+func FromContext(ctx context.Context, logger *logging.ZapEventLogger) *logging.ZapEventLogger {
+	if logger, ok := ctx.Value("loggerKey").(*logging.ZapEventLogger); ok {
+		return logger
+	}
+	return logger
+}
+
+func WithLogger(ctx context.Context, logger *logging.ZapEventLogger) context.Context {
+	return context.WithValue(ctx, loggerKey, logger)
+}
+
+func NewFileLogger(level string, name string) (*logging.ZapEventLogger, error) {
```

Comment:
Just to see if I've understood this: the reason why you use low-level `zap` insteado of using directly the ipfs-logger and setting `GOLOG_OUTPUT` is because you want to output mining logs into an independent file, right?

Reply:
Yes, correct. `GOLOG_OUTPUT` doesn't allow to specify the concrete file for the target log stream.